### ISSUE TITLE
affine transform issue (divided by zero)

### ISSUE THREAD

**Issue by AtousaTorabi** [anatolix/keras_Realtime_Multi-Person_Pose_Estimation#28]
Hi,
Thanks for great work to do augmentation without server. I get divided error by zero when training at affine function in this line:
scale_size = config.transform_params.target_dist / (scale_self * self.scale)
seems scale might be zero sometimes.

Thank you!

**Comment by abinjoabraham** [comment]
Even I am facing the same issue now. I will post if I find a solution. @AtousaTorabi

**Comment by murrayLuke** [comment]
Interesting that could explain the NaN and inf values I've been seeing.

**Comment by abinjoabraham** [comment]
Still I am stuck in this situation .

**Comment by murrayLuke** [comment]
I don't think it would be too hard to write a one liner and change the scale size if it is 0.

**Comment by abinjoabraham** [comment]
I have wrote a simple if statement to skip the 0 of the scale size. But not sure what value I need to give to scale size if it is 0.

**Comment by murrayLuke** [comment]
Try 1 since that should give no scaling whatsoever.

**Comment by abinjoabraham** [comment]
Tried the same and it is running. But the loss is going nan.

**Comment by anatolix** [comment]
neither scale_self nor self.scale shouldn't be zero. scale_self  taken from hdf5, check dataset generation, may be we need to put assert there.

**Comment by abinjoabraham** [comment]
Issue is not anymore for me. I put an assert in the dataset generation file and also changed the first learning rate of the epochs.

**Comment by YicongHong** [comment]
@abinjoabraham I tried to add an if condition in the data generation file to avoid divide by zero:
              if (img_anns[p]["bbox"][3]) == 0:    pers["scale_provided"] = 1
but the loss is going nan, I am wondering how exactly should I modify the dataset generation file and change the first learning rate?

**Comment by YicongHong** [comment]
btw, why should we use the height of the bounding box divide by 368.0 as the scale?
as in the data generation file:
pers["scale_provided"] = img_anns[p]["bbox"][3] / 368.0

**Comment by abinjoabraham** [comment]
@YicongHong Exact lines are below 

        if pers["scale_provided"]==0:      
            pers["scale_provided"] = 0.2

Also regarding the epoch learning rate, I reverted back to the original line of codes. As it was not making any issues.
